compiler/optimizer/ValuePropagationCommon.cpp

```diff
@@ -3730,6 +3734,10 @@ void OMR::ValuePropagation::transformConverterCall(TR::TreeTop *callTree)
 
    TR::Node *child1 = TR::Node::create(TR::iadd, 2, srcOff, len);
    TR::Node *child2 = TR::Node::create(TR::arraylength, 1, srcObjNode);
+   // the source for the decompressed string version of this in StirngCoding takes a byte array
+   // used as a char array so convert the size appropriately for the check >> 1
+   if (rm == TR::java_lang_StringCoding_implEncodeISOArray)
+      child2 = TR::Node::create(TR::ishr, 2, child2, TR::Node::iconst(child2, 0));
```

Comment:
Is this supposed to be a shift right by 1? Is the `TR::Node::iconst(child2, 0)` here correct? Also StringCoding is spelled incorrectly.

Reply:
fixed